src/extractor/node_based_graph_factory.cpp

```diff
@@ -150,20 +169,42 @@ void NodeBasedGraphFactory::CompressAnnotationData()
         for (EdgeID nbg_edge_id : compressed_output_graph.GetAdjacentEdgeRange(nbg_node_u))
         {
             auto const &edge = compressed_output_graph.GetEdgeData(nbg_edge_id);
-            annotation_mapping[edge.annotation_data] = 0;
+            auto res = unique_annotations.insert(edge.annotation_data);
+            annotation_mapping[edge.annotation_data] = *res.first;
```

Comment:
Since we are looking at the ID's in order, we could directly assign the correct mapping here.

Reply:
No, but we are assigning them based on valid/used entries. Before we were only using a vector to remove duplicated entries. But you are hashing all entries anyhow, so you could directly assign a final value, if you are do full re-mapping anyhow. That way we could remove prefix sum computations and do a single re-ordering step.